lib/ash_ai/actions/prompt.ex

```diff
@@ -57,6 +57,24 @@ defmodule AshAi.Actions.Prompt do
   The first argument to `prompt/2` is the `LangChain` model. It can also be a 2-arity function which will be invoked
   with the input and the context, useful for dynamically selecting the model.
 
+  ## Dynamic Configuration (using 2-arity function)
+
+  For runtime configuration (like using environment variables), pass a function
+  as the first argument to `prompt/2`:
+
+      run prompt(
+        fn _input, _context ->
+          LangChain.ChatModels.ChatOpenAI.new!(%{
+            model: "gpt-4o",
+            api_key: System.get_env("OPENAI_API_KEY"),
```

Comment:
```suggestion
            # this can also be configured in application config, see langchain docs for more.
            api_key: System.get_env("OPENAI_API_KEY"),
```